cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -129,18 +136,19 @@ Result<ExtensionSet> ExtensionSet::Make(
     return Status::Invalid("Type ", type_ids[i].uri, "#", type_ids[i].name, " not found");
   }
 
-  set.functions_.reserve(function_ids.size());
+  // set.functions_.reserve(function_ids.size());
 
   for (unsigned int i = 0; i < static_cast<unsigned int>(function_ids.size()); ++i) {
     if (function_ids[i].empty()) continue;
     RETURN_NOT_OK(set.CheckHasUri(function_ids[i].uri));
-
-    if (auto rec = registry->GetFunction(function_ids[i])) {
-      set.functions_[i] = {rec->id, rec->function_name};
-      continue;
-    }
-    return Status::Invalid("Function ", function_ids[i].uri, "#", function_ids[i].name,
-                           " not found");
+    set.functions_[i] = {{function_ids[i].uri, function_ids[i].name}, "dummy"};
+
+    // if (auto rec = registry->GetFunction(function_ids[i])) {
+    //   set.functions_[i] = {rec->id, rec->function_name};
+    //   continue;
+    // }
+    // return Status::Invalid("Function ", function_ids[i].uri, "#", function_ids[i].name,
+    //                        " not found");
```

Comment:
should we remove this or intended to use?

Reply:
I've uncommented this and merged `FunctionMapping` into `ExtensionIdRegistry`.